specs/mgt-taxonomy-picker.md

```diff
@@ -0,0 +1,87 @@
+# mgt-taxonomy-picker
+
+## Overview
+The taxonomy picker is a component that can query the Microsoft Graph API for Taxonomy and render a dropdown control allowing selection of **a single** term based on the specified term set id and a combination of the specified term set id and the specified term id. 
+
+> **Note**
+> This capability used to support a single selection coming from the term store. This will be reviewed in the future.  
+
+## User Scenarios
+
+Let's assume a term store with the following structure:
+
+![mgt-taxonomy-pikcer-structure](./images/mgt-taxonomy-pikcer-structure.png)
+
+### Scenario 1 - Select a term from a term set
+
+Shows the terms marked with a red box in the image above.
+
+![mgt-taxonomy-pikcer-termset](./images/mgt-taxonomy-pikcer-termset.png)
+
+### Scenario 2 - Select a term from a term in a term set
+
+Shows the terms marked with a green box in the image above.
+
+![mgt-taxonomy-pikcer-term](./images/mgt-taxonomy-pikcer-term.png)
+
+
+### Scenario 3 - Select a term from a term in a term set in a different language
+
+Shows French labels of the terms marked with a green box in the image above.
+
+![mgt-taxonomy-pikcer-locale-term](./images/mgt-taxonomy-pikcer-locale-term.png)
+
+
+## Proposed Solution
+
+### Examples
+
+#### Show the first level children of the term set with the specified id.
+
+`<mgt-taxonomy-picker termset-id="138a652e-7f23-46f6-b480-13da2308c235"></mgt-taxonomy-picker>`
+
+#### Show the first level children of the term with the specified id under the term set with the specified id.
+
+`<mgt-taxonomy-picker termset-id="138a652e-7f23-46f6-b480-13da2308c235" term-id="a56caeb7-3b7d-4d22-93a9-0232e12905f6"></mgt-taxonomy-picker>`
+
+#### Show the first level children of the term with the specified id under the term set with the specified id in a different language.
+
+`<mgt-taxonomy-picker termset-id="138a652e-7f23-46f6-b480-13da2308c235" term-id="a56caeb7-3b7d-4d22-93a9-0232e12905f6" locale="fr-FR"></mgt-taxonomy-picker>`
+
+#### Cache the response from the API call for 50 seconds.
+
+`<mgt-taxonomy-picker termset-id="138a652e-7f23-46f6-b480-13da2308c235" cache-enabled="true" cache-invalidation-period="50000"></mgt-taxonomy-picker>`
+
+## Properties and Attributes
```

Comment:
@anoopt could we have a `level` property which would allow you to specify a number of levels deep within a term set that you can display. It will be used with a `termset-id` and have a default value of 2. This way, you can display the entire termset as shown without specifying a term-id. What do you think? :

![image](https://user-images.githubusercontent.com/8081536/229109005-bda6718c-8231-42d8-a983-ad35ebfacce0.png)

Reply:
@musale - Since the data [returned by Microsoft Graph](https://learn.microsoft.com/en-us/graph/api/resources/termstore-term?view=graph-rest-1.0) for a term (please see below), doesn't have any details related to the parent term or child term, I don't think we can specify the `level` property. 

```json
{
  "@odata.type": "#microsoft.graph.termStore.term",
  "id": "String (identifier)",
  "labels": [
    {
      "@odata.type": "microsoft.graph.termStore.localizedLabel"
    }
  ],
  "createdDateTime": "String (timestamp)",
  "lastModifiedDateTime": "String (timestamp)",
  "descriptions": [
    {
      "@odata.type": "microsoft.graph.termStore.localizedDescription"
    }
  ],
  "properties": [
    {
      "@odata.type": "microsoft.graph.keyValue"
    }
  ]
}
```

However, if all the terms under a termset need to be listed, then we can use the `/termStore/sets/{termset-id}/terms` endpoint. This endpoint works only for a termset and not a term.

Reply:
Oh I see, thank you 😄